docker-compose.yml

```diff
@@ -0,0 +1,64 @@
+version: "3"
+
+# SPECIAL NOTES:
+# 1.  GEM_HOME is set to the tmp/gem_home directory in this app.
+#     Bundler is installed here.
+#     If you delete Bundler from this directory, you will get an error
+#     message about not having the correct version of bundler when you
+#     run "bundle install" or try to do just about anything else.
+# 2.  BUNDLE_PATH is set to the tmp/bundle directory in this app.
+#     The products of "bundle install" are saved here.  If you delete
+#     this directory, subsequent executions of the "bundle install"
+#     process will take longer.
+
+services:
+
+  mailcatcher:
+    image: yappabe/mailcatcher
+    ports:
+      - "1025:1025"
+      - "1080:1080"
+
+  database:
+    image: postgres
+    env_file:
+      - .docker-env/development/database
+    volumes:
+      - db_data:/var/lib/postgresql/data
+      - ./init.sql:/docker-entrypoint-initdb.d/init.sql
+    ports:
+      - "15432:5432"
+
+  rails:
+    container_name: pie_rails
+    build: .
+    volumes:
+      - .:/home/winner/myapp
```

Comment:
* /home/winner/myapp is the WORKDIR specified in the Dockerfile (Dockerfile-template).
* "winner" is the username in the Docker environment.  I prefer to keep the username and the project name different, because using the same thing for both can lead to confusion if I need to troubleshoot/debug something.
* I set up the Rails Docker container to be run by the regular user as default.  Doing development as root is generally frowned on, yet it seems that most uses of Docker involve running the Docker containers as root.
* I keep the working directory in the Docker container one level below the user home directory, because making the working directory and user home directory one and the same contaminates the repository with user configuration files and directories.

Reply:
out of curiosity is `/home/username` the typical setup for Linux boxes?  For some reason I thought it was `/usr/username` or `/username` but I might be misremembering.  I understand wanting to keep it separate for debugging, can we use something like `docker` so it's clearer where it's coming from rather than `winner`?  I'm fine if you want to keep the run-docker-as-regular-user behavior though, that's fine.

Reply:
Yes, /home/username is the typical Linux setup.  I don't want to use "docker" as the username, because that could be a source of confusion.  If you wish, I could change the username from "winner" to something like "docker1" or "docker_user".